auslib/test/test_db.py

```diff
@@ -2698,6 +2710,33 @@ def testProceedIfNotReadOnly(self):
         self.releases.t.update(values=dict(read_only=True, data_version=2)).where(self.releases.name == "a").execute()
         self.assertRaises(ReadOnlyError, self.releases._proceedIfNotReadOnly, 'a')
 
+    def testGetRulesMatchingQueryChannelImprovedGlobbing(self):
+        # To ensure length of ruleChannel is >=3
+        expected = []
+        rules = self.rules.getRulesMatchingQuery(
+            dict(name='', product='', version='3.0', channel='releasetest',
+                 buildTarget='', buildID='', locale='', osVersion='', distribution='',
+                 distVersion='', headerArchitecture='',
+                 force=False, queryVersion=3,
+                 ),
+            fallbackChannel='releasetest'
+        )
+        rules = self._stripNullColumns(rules)
+        self.assertEquals(rules, expected)
+
+        # To ensure globbing at end only
+        expected = [dict(rule_id=6, priority=100, backgroundRate=100, channel='r*test*', update_type='z', data_version=1)]
```

Comment:
As I mentioned earlier, please make sure each test case is in its own function - otherwise they may not always be run.